### PR TITLE
dev/core#3994 fix saving of auto renew option in membership type block for contribution page configuration

### PR DESCRIPTION
**Pull request by kurund**

### CODE REVIEW

**Comment by stesi561** [comment]
I just tried to replicate this on https://dmaster.demo.civicrm.org/civicrm/admin/member/membershipType?reset=1 and was not able to replicate the failure. The screen looks like it has changed from the screenshot in  https://lab.civicrm.org/dev/core/-/issues/3994 @kurund can you confirm this is still an issue and if so please provide urls of the page you start from to edit the membership types please. Otherwise this can be closed.

**Comment by stesi561** [comment]
Re Test this please

**Comment by stesi561** [comment]
@kurund you might need to rebase the above - looks like a merge conflict with https://github.com/civicrm/civicrm-core/ - I think you are removing the relevant line anyway.

**Comment by kurund** [comment]
@stesi561 

Apologies for the delayed response. I have rebased the PR. Issue still persist.

File: CRM/Member/Form/MembershipBlock.php [hunk 1]
```diff
@@ -126,32 +126,59 @@ public function buildQuickForm() {
...
 
       //CRM-15573
-      if (!empty($params['id'])) {
-        $params['membership_types'] = serialize($membershipRequired);
-        CRM_Member_BAO_MembershipBlock::writeRecord($params);
+      // check if we have any auto-renew membership types and determine if the settings needs to be
+      // auto updated
+      if (!empty($autoRenewMembershipTypes)) {
+        // get the existing membership types for this page
+        $membershipBlocks = \Civi\Api4\MembershipBlock::get()
```
Comment (colemanw) [inline CRM/Member/Form/MembershipBlock.php:155]: @kurund did you mean to have permission checks in this api call?

File: CRM/Member/Form/MembershipBlock.php [hunk 2]
```diff
@@ -126,32 +126,59 @@ public function buildQuickForm() {
...
+            }
+            else {
+              $updatedMembershipTypes[$mtype] = $value;
+            }
+          }
+
+          // let's update only if need to which is only when membership type auto renew
+          // option changes in the membership type configuration
+          if ($membershipBlocks['membership_types'] != $updatedMembershipTypes) {
+            \Civi\Api4\MembershipBlock::update()
```
Comment (colemanw) [inline CRM/Member/Form/MembershipBlock.php:176]: @kurund also permission checks in this api call?

File: CRM/Member/Form/MembershipBlock.php (hunk 1 above)
Comment (colemanw) [reply CRM/Member/Form/MembershipBlock.php:155]: Ok

File: CRM/Member/Form/MembershipBlock.php (hunk 2 above)
Comment (kurund) [reply CRM/Member/Form/MembershipBlock.php:176]: same reason as above,

Comment (colemanw) [reply CRM/Member/Form/MembershipBlock.php:176]: That's fine

**Comment by colemanw** [comment]
@stesi561 would you be able to retest this please now that it's rebased?

**Comment by eileenmcnaughton** [comment]
@kurund I'm closing this out cos it is stale but I took a look at whether it could be revived & came up with an alternate approach that moves the problematic handling to a place that I think is more appropriate https://github.com/civicrm/civicrm-core/pull/28387

**Pull request closed without merging**